src/utils/pureComments.ts

```diff
@@ -0,0 +1,46 @@
+// @ts-ignore
+import walk from 'acorn/dist/walk';
+import * as ESTree from 'estree';
+import { CommentDescription } from '../Module';
+
+function forEachNodeAfterComment(
+	ast: ESTree.Program,
+	commentNodes: CommentDescription[],
+	handleNode: (node: ESTree.Node) => void
+): void {
+	let commentIdx = 0;
+
+	checkCommentsBeforeNode(<any>ast);
+
+	function checkCommentsBeforeNode(node: ESTree.Node & { start: number; end: number }) {
+		if (commentIdx === commentNodes.length || commentNodes[commentIdx].end > node.end) return;
+		let isFound = false;
+		while (commentIdx < commentNodes.length && node.start >= commentNodes[commentIdx].end) {
```

Comment:
There was a (rare) issue where multiple comments belonging to the same node were actually attributed to subsequent node. This loop fixes this.

Reply:
Good catch.